base36.go

```diff
@@ -0,0 +1,158 @@
+/*
+
+Package base36 provides a reasonably fast implementation of a binary base36 codec.
+
+*/
+package base36
+
+// Simplified code based on https://godoc.org/github.com/mr-tron/base58
+// which in turn is based on https://github.com/trezor/trezor-crypto/commit/89a7d7797b806fac
+
+import (
+	"fmt"
+	"math"
+)
+
+const UcAlphabet = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const LcAlphabet = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+var encExpansion int
+var revAlphabet [256]uint64
+
+func init() {
+	for i := range revAlphabet {
+		revAlphabet[i] = 36
```

Comment:
"invalid" sentinel? Can we const it?

Reply:
Isn't `36` already a const...?
It's being checked here: https://github.com/multiformats/go-base36/blob/11f3b26a8e0e9068d38d6fa365d487dd00c6d057/base36.go#L122

( I also need to add a check back I dropped stupidly 🤦 )

Reply:
I mean define a constant somewhere. It took me a while to figure out why we were using 36. If it were just something like "invalid", that would be easier to understand.

Reply:
Duh...
https://github.com/multiformats/go-base36/blob/438fc2b1a7f50241be57e5d9801441c5050fa9fa/base36.go#L18-L24
+
https://github.com/multiformats/go-base36/blob/438fc2b1a7f50241be57e5d9801441c5050fa9fa/base36.go#L118